### ISSUE TITLE
Deferred package fetching

### ISSUE THREAD

**Issue by dabrahams** [jwiegley/use-package#1040]
When I arrive on a new machine and fire up emacs to, say, edit an Org file—or after I've made quite a few configuration updates—I don't want to have to wait for myriad packages to install.  

What I'd like, for many packages, is to say `:ensure`, but do it async. https://github.com/conao3/feather.el could be useful for the async part, but with use-package, there seems to be no way to get the `:config` clause to be done as an `eval-after-load`.

**Comment by skangas** [comment]
I think this would be a useful feature. Patches welcome.

**Comment by dabrahams** [comment]
Now that I look, I wonder if this is the semantics provided by `:defer 0`?

**Comment by skangas** [comment]
> Now that I look, I wonder if this is the semantics provided by `:defer 0`?

I don't think so, not unless I misunderstood your original feature request. On my machine, this is what `(use-package foo :defer 0)` expands to:

```
(progn
  (use-package-ensure-elpa 'foo '(t) 'nil)
  (run-with-idle-timer 0 nil (function require) 'foo nil t))
```

In other words, the package is still installed immediately.
